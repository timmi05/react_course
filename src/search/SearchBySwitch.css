.search-by-switch{
    display: flex;
}

.search-by-switch-label{
    padding-right: 15px;
    display: flex;
    justify-content: center;
    flex-direction: column;
}

.search-by-switch-button{
    font-size: 15px;
    padding: 10px 26px;
}