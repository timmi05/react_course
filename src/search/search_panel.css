.search-panel {
    margin: 40px;
}

.search-panel-label{
    margin-bottom: 30px;
    font-size: 40px;
}

.search-panel-main{
    padding-bottom: 15px;
    display: flex;
}

.search-panel-input{
    width: 85%;
    margin-right: 20px;
    height: 40px;
    font-size: 28px;
}

.search-panel-button{
    font-size: 20px;
    padding: 10px 35px;
}