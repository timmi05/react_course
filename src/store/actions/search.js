export const search = (value, searchBy) => ({
    type: 'SEARCH_MOVIES',
    payload: {
        value: value,
        searchBy: searchBy
    }
});

export const moveToSearch = () => ({
    type: 'MOVE_TO_SEARCH'
});
