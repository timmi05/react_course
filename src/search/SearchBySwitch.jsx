import React, { useState } from 'react';
import { Button } from '../component/Button'
import './SearchBySwitch.css';

export const SearchBySwitch = () => {

    const [disabled, setDisabled] = useState(false);

    const SearchByTitle = () => {
        setDisabled(true);
    };

    const SearchByGenre = () => {
        setDisabled(false);
    };

    return (
        <div className='search-by-switch'>
            <span className='search-by-switch-label'>SEARCH BY </span>
            <div className = 'search-by-switch-buttons'>
                <Button
                label = 'TITLE'
                classes = 'search-by-switch-button search-by-switch-left-button'
                onClick = { SearchByTitle }
                disabled = { disabled }
                />
                <Button
                label = 'GENRE'
                classes = 'search-by-switch-button search-by-switch-right-button'
                onClick = { SearchByGenre }
                disabled = { !disabled }
                />
            </div>
        </div>
    );
};
