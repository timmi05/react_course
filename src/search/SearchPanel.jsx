import React, { useState }  from 'react';
import { SearchBySwitch } from './SearchBySwitch';
import { Button } from '../component/Button'
import './SearchPanel.css';

export const SearchPanel = () => {

    const [value, setValue] = useState('');
    const [disabled, setDisabled] = useState(false);

    const handleChange = event => {
      setValue(event.target.value);
    };

    const handleSubmit = () => {
      setDisabled(true);
    };

    return (
        <div className='search-panel'>
           <div className='search-panel-label'>FINED YOUR MOVIE</div>
           <div className='search-panel-main'>
               <input className='search-panel-input' type='text' value={ value } onChange={ handleChange } />
               <Button
               label = 'SEARCH'
               classes = 'search-panel-button'
               onClick = { handleSubmit }
               disabled = { disabled }
               />
           </div>
           <SearchBySwitch/>
        </div>
    );
};
