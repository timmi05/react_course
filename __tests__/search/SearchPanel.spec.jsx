import React from 'react';
import { mount } from 'enzyme';
import { SearchPanel } from '../../src/search/SearchPanel';

describe('SearchPanel rendering', () => {
    const component = mount(<SearchPanel/>);
    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('simulate change of input', () => {
        component.find('input').simulate('change');
        expect(component).toMatchSnapshot();
    });

    it('simulate click on button SEARCH', () => {
        component.find({label: 'SEARCH'}).simulate('click');
        expect(component.find({className:'button search-panel-button'})
            .prop('disabled')).toBe(true);
        expect(component).toMatchSnapshot();
    });
});
