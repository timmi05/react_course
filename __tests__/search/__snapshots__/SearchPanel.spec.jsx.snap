// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel rendering renders correctly 1`] = `
<SearchPanel>
  <div
    className="search-panel"
  >
    <div
      className="search-panel-label"
    >
      FINED YOUR MOVIE
    </div>
    <div
      className="search-panel-main"
    >
      <input
        className="search-panel-input"
        onChange={[Function]}
        type="text"
        value=""
      />
      <Button
        classes="search-panel-button"
        disabled={false}
        label="SEARCH"
        onClick={[Function]}
      >
        <button
          className="button search-panel-button"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          SEARCH
        </button>
      </Button>
    </div>
    <SearchBySwitch>
      <div
        className="search-by-switch"
      >
        <span
          className="search-by-switch-label"
        >
          SEARCH BY 
        </span>
        <div
          className="search-by-switch-buttons"
        >
          <Button
            classes="search-by-switch-button search-by-switch-left-button"
            disabled={false}
            label="TITLE"
            onClick={[Function]}
          >
            <button
              className="button search-by-switch-button search-by-switch-left-button"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              TITLE
            </button>
          </Button>
          <Button
            classes="search-by-switch-button search-by-switch-right-button"
            disabled={true}
            label="GENRE"
            onClick={[Function]}
          >
            <button
              className="button search-by-switch-button search-by-switch-right-button"
              disabled={true}
              onClick={[Function]}
              type="submit"
            >
              GENRE
            </button>
          </Button>
        </div>
      </div>
    </SearchBySwitch>
  </div>
</SearchPanel>
`;

exports[`SearchPanel rendering simulate change of input 1`] = `
<SearchPanel>
  <div
    className="search-panel"
  >
    <div
      className="search-panel-label"
    >
      FINED YOUR MOVIE
    </div>
    <div
      className="search-panel-main"
    >
      <input
        className="search-panel-input"
        onChange={[Function]}
        type="text"
        value=""
      />
      <Button
        classes="search-panel-button"
        disabled={false}
        label="SEARCH"
        onClick={[Function]}
      >
        <button
          className="button search-panel-button"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          SEARCH
        </button>
      </Button>
    </div>
    <SearchBySwitch>
      <div
        className="search-by-switch"
      >
        <span
          className="search-by-switch-label"
        >
          SEARCH BY 
        </span>
        <div
          className="search-by-switch-buttons"
        >
          <Button
            classes="search-by-switch-button search-by-switch-left-button"
            disabled={false}
            label="TITLE"
            onClick={[Function]}
          >
            <button
              className="button search-by-switch-button search-by-switch-left-button"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              TITLE
            </button>
          </Button>
          <Button
            classes="search-by-switch-button search-by-switch-right-button"
            disabled={true}
            label="GENRE"
            onClick={[Function]}
          >
            <button
              className="button search-by-switch-button search-by-switch-right-button"
              disabled={true}
              onClick={[Function]}
              type="submit"
            >
              GENRE
            </button>
          </Button>
        </div>
      </div>
    </SearchBySwitch>
  </div>
</SearchPanel>
`;

exports[`SearchPanel rendering simulate click on button SEARCH 1`] = `
<SearchPanel>
  <div
    className="search-panel"
  >
    <div
      className="search-panel-label"
    >
      FINED YOUR MOVIE
    </div>
    <div
      className="search-panel-main"
    >
      <input
        className="search-panel-input"
        onChange={[Function]}
        type="text"
        value=""
      />
      <Button
        classes="search-panel-button"
        disabled={true}
        label="SEARCH"
        onClick={[Function]}
      >
        <button
          className="button search-panel-button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          SEARCH
        </button>
      </Button>
    </div>
    <SearchBySwitch>
      <div
        className="search-by-switch"
      >
        <span
          className="search-by-switch-label"
        >
          SEARCH BY 
        </span>
        <div
          className="search-by-switch-buttons"
        >
          <Button
            classes="search-by-switch-button search-by-switch-left-button"
            disabled={false}
            label="TITLE"
            onClick={[Function]}
          >
            <button
              className="button search-by-switch-button search-by-switch-left-button"
              disabled={false}
              onClick={[Function]}
              type="submit"
            >
              TITLE
            </button>
          </Button>
          <Button
            classes="search-by-switch-button search-by-switch-right-button"
            disabled={true}
            label="GENRE"
            onClick={[Function]}
          >
            <button
              className="button search-by-switch-button search-by-switch-right-button"
              disabled={true}
              onClick={[Function]}
              type="submit"
            >
              GENRE
            </button>
          </Button>
        </div>
      </div>
    </SearchBySwitch>
  </div>
</SearchPanel>
`;
